singlestoredb/fusion/handlers/files.py

```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/env python3
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+from .. import result
+from ..handler import SQLHandler
+from ..result import FusionSQLResult
+from .utils import dt_isoformat
+from .utils import get_file_space
+
+
+class ShowFilesHandler(SQLHandler):
+    """
+    SHOW <file-location> FILES
```

Comment:
Unfortunately, commands like this are going to cause collisions with other commands. Fusion SQL needs a unique prefix of literal keywords to know that the command is actually a Fusion SQL command and not a real database command. In this case the only literal keyword prefix is `SHOW` which overlaps with other command here and in the database. To get around this, you'll need to give each location its own command like `SHOW PERSONAL FILES` and `SHOW SHARED FILES`. If you write the `run` method to look at `self.command_key`, it will contain the prefix keywords such as `('SHOW', 'SHARED', 'FILES')` then you could write a parent class with the generic `run` method and subclass it with the variants containing the different docstrings for the different grammars and documentation.

Reply:
Thank you for noticing this (again 🙃 ))! I have refactored it to address this issue. I used the `params` dictionary instead of `self.command_key` to pass the file location to the parent class. Please let me know if you are okay with this change.